polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1072,17 +1081,19 @@ where
 		.insert_advertisement(
 			relay_parent,
 			relay_parent_mode,
+			use_non_prospective_chains_mode,
 			candidate_hash,
 			&state.implicit_view,
 			&state.active_leaves,
 		)
 		.map_err(AdvertisementError::Invalid)?;
+
 	if !per_relay_parent.collations.is_seconded_limit_reached(relay_parent_mode) {
 		return Err(AdvertisementError::SecondedLimitReached)
 	}
 
 	if let Some((candidate_hash, parent_head_data_hash)) = prospective_candidate {
-		let is_seconding_allowed = !relay_parent_mode.is_enabled() ||
+		let is_seconding_allowed = !use_non_prospective_chains_mode ||
```

Comment:
This bit doesn't quite make sense to me.

Previously, the candidate backing system was checking the parent head data hash.

We should still do the `can_second` check even if the collator version is old (not rely on them to be honest), but just make sure that the `parent_head_data_hash` is the on-chain included block

Reply:
Actually I don't get this at all: We only get here if prospective_candidate is `Some`, which implies that `use_non_prospective_chains_mode` is [false](https://github.com/paritytech/polkadot-sdk/blob/5291601ff4948c9181e4f8edddb00c18fab7ba19/polkadot/node/network/collator-protocol/src/validator_side/mod.rs#L1059). !use_non_prospective_chains_mode will always be `true` here.